### PR TITLE
Feature/table

### PR DESCRIPTION
**Pull request by hesamhabibih**

### CODE REVIEW

File: app/[locale]/page.tsx
Comment (amirkh330) [inline app/[locale]/page.tsx:1]: clear test files in pages

File: components/Table/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
+import * as React from 'react'
+import { useState, useRef, useMemo, useEffect } from 'react'
+// import DataService from "./services/TutorialService";
```
Comment (amirkh330) [inline components/Table/Table.tsx]: remove useless line

File: components/Table/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+import { useTable } from 'react-table'
+import Pagination from '@material-ui/lab/Pagination'
+
+interface ITableProps {
+  history: any
```
Comment (amirkh330) [inline components/Table/Table.tsx]: use the correct type instead any

File: components/Table/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+  DataService: any
+  pageSizes: any
+  columns: any
+  costumeClassName: any
+}
+const Table = (props: ITableProps) => {
+  const [tutorials, setTutorials] = useState([])
+  const [searchTitle, setSearchTitle] = useState('')
+  const tutorialsRef: any = useRef()
```
Comment (amirkh330) [inline components/Table/Table.tsx]: use correct type 
I think you can use this type
React.RefObject<HTMLElement>

File: components/Table/Table.tsx [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+  const [page, setPage] = useState(1)
+  const [count, setCount] = useState(0)
+  const [pageSize, setPageSize] = useState(3)
+
+  // const pageSizes = [3, 6, 9];
```
Comment (amirkh330) [inline components/Table/Table.tsx]: remove this line

File: components/Table/http-common.ts
Comment (amirkh330) [inline components/Table/http-common.ts:1]: it's necessary?

File: components/Table/Table.tsx (hunk 3 above)
Comment (hesamhabibih) [reply components/Table/Table.tsx]: I can't find better way for editing type of prop(tutorialsRef) but refactor this code again and again thanks .I change name of it to suitable name in code

File: components/Table/http-common.ts
Comment (hesamhabibih) [reply components/Table/http-common.ts:1]: I think  yes. we can keep this code for trace  component code in other levels . please keep it. I remove this next

**Review by amirkh330 — APPROVED** [review]

File: components/Table/services/TutorialService.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import http from './http-common'
+
+const getAll = (params: any | undefined) => {
+  debugger
```
Comment (saralabafi) [inline components/Table/services/TutorialService.ts:4]: remove debugger

**Review by saralabafi — APPROVED** [review]

**Pull request merged**